core/trino-main/src/test/java/io/trino/sql/analyzer/TestAnalyzer.java

```diff
@@ -3890,6 +3890,40 @@ public void testInvalidDelete()
         assertFails("DELETE FROM v1 WHERE a = 1")
                 .hasErrorCode(NOT_SUPPORTED)
                 .hasMessage("line 1:1: Deleting from views is not supported");
+        assertFails("DELETE FROM mv1")
+                .hasErrorCode(NOT_SUPPORTED)
+                .hasMessage("line 1:1: Deleting from materialized views is not supported");
+    }
+
+    @Test
+    public void testInvalidUpdate()
+    {
+        assertFails("UPDATE foo SET bar = 'test'")
+                .hasErrorCode(TABLE_NOT_FOUND)
+                .hasMessage("line 1:1: Table 'tpch.s1.foo' does not exist");
+        assertFails("UPDATE v1 SET a = 2")
+                .hasErrorCode(NOT_SUPPORTED)
+                .hasMessage("line 1:1: Updating through views is not supported");
+        assertFails("UPDATE v1 SET a = 2 WHERE a = 1")
```

Comment:
is there a purpose to have both test `UPDATE v1 SET a = 2` and `UPDATE v1 SET a = 2 WHERE a = 1`?

Reply:
No, actually not. I was wondering as well while I adapted the method from `testInvalidUpdate()`.